### DISCUSSION TITLE
Add support for `volatile` prop in vector source

### DISCUSSION THREAD

**Discussion by grillorafael** [rnmapbox/maps#3351] [Ideas]
Hi!

Currently both in v10 and v11 mapbox support the option to pass volatile true/false to a vector source.

This controls whether or not that source should be cached to disk.

I was looking to get that implemented - thoughts ?

**Comment by mfazekas** [comment]
👍 sounds like a straightforward and useful change

**Comment by marius-at-atb** [comment]
This would be very useful. Is it planned to be supported soon?

**Reply by marius-at-atb** [reply]
Also I wonder – instead of just choosing immediate invalidation or infinite validity, is it possible to specify a time interval for when stale data is still shown?
Ideally using native functionality instead of updating the js filter and adding timestamps to every feature from the tile server. I suspect that would severely impact performance 🤔 

In react-query terms - the max-age header allows control over staleTime, but what about cacheTime, can that be specified somehow as well?
Would be awesome to have, and even together with a prop, e.g. `isStale` on the feature, so that one can add visual queues.
